applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_element.cpp

```diff
@@ -830,14 +830,13 @@ void UPwSmallStrainElement<TDim, TNumNodes>::CalculateMaterialStiffnessMatrix(Ma
 
 template <unsigned int TDim, unsigned int TNumNodes>
 void UPwSmallStrainElement<TDim, TNumNodes>::CalculateAndAddGeometricStiffnessMatrix(
-    MatrixType& rLeftHandSideMatrix, const ElementVariables& rVariables, unsigned int GPoint)
+    MatrixType& rLeftHandSideMatrix, const Vector& rStressVector, const Matrix& rGradNpt, double IntegrationCoefficient)
 {
     KRATOS_TRY
 
-    const Matrix stress_tensor = MathUtils<double>::StressVectorToTensor(mStressVector[GPoint]);
-    const Matrix reduced_Kg_matrix =
-        prod(rVariables.GradNpT, rVariables.IntegrationCoefficient *
-                                     Matrix(prod(stress_tensor, trans(rVariables.GradNpT)))); // to be optimized
+    Matrix stress_tensor = MathUtils<double>::StressVectorToTensor(rStressVector);
+    Matrix reduced_Kg_matrix =
+        prod(rGradNpt, IntegrationCoefficient * Matrix(prod(stress_tensor, trans(rGradNpt))));
```

Comment:
The formula is: integral (dN/dx)^T sigma (dN/dx) dV
( give or take the place of the transpose ^T.
the int .. dV is the scalar multiplication with IntegrationCoefficient. That multiplication could for clarity be outside the outermost prod() or for efficiency on the smallest matrix used. ( don't know whether that is rStressVector or rGradNpt )
It is clear to define the intermediate variable stress_tensor, but for me it is equally clear to write:
Matrix reduced_Kg_matrix =
        prod(rGradNpt, Matrix(prod(MathUtils<double>::StressVectorToTensor(rStressVector), trans(rGradNpt)))) * IntegrationCoefficient;

Reply:
Agreed, that way it reads more like the integral from the formulation, changed